### PR TITLE
Adopt Edward wrapper

### PR DESCRIPTION
**Pull request by fonnesbeck**
As suggested in #1626 this PR adopts the wrapper for fitting PyMC3 models using Edward. Accordingly, I've renamed the wrapper from `PyMC3Model` to `EdwardModel`, and put it in a subdirectory called `external` to accommodate this and future interfaces to other packages.

I also adopted the Beta-Bernoulli example from Edward. Do we want a test for this, or is the example sufficient?

### CODE REVIEW

**Comment by ferrine** [comment]
Looks interesting. Does it mean that installation requires lots of dependencies?

**Comment by fonnesbeck** [comment]
Just Edward (and its TensorFlow dependency, of course), which I have listed in the `extras_require` field in `setup`, so users can ignore it if they wish.

File: pymc3/examples/edward_beta_bernoulli.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+model = EdwardModel(pm_model)
+
+data = {x_obs: np.array([0, 1, 0, 0, 0, 0, 0, 0, 0, 1])}
+
+qp_a = tf.nn.softplus(tf.Variable(tf.random_normal([])))
+qp_b = tf.nn.softplus(tf.Variable(tf.random_normal([])))
+qp = Beta(a=qp_a, b=qp_b)
+
+inference = ed.KLqp({'p': qp}, data, model)
+inference.run(n_iter=500)
```
Comment (ferrine) [inline pymc3/examples/edward_beta_bernoulli.py]: New line

**Comment by springcoil** [comment]
This looks fine to me. Perhaps a test though for the Bernoulli example.

File: pymc3/external/edward.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        # Set data placeholders in PyMC3 model (Theano shared
+        # variable) to their realizations (NumPy array).
+        for key, value in zip(self.xs_keys, xs_values):
+            key.set_value(value)
+
+        # Calculate model's log density using a dictionary of latent
+        # variables.
+        z = {key: value for key, value in zip(self.zs_keys, zs_values)}
+        lp = self.model.fastlogp(z)
+        return lp.astype(np.float32)
```
Comment (twiecki) [inline pymc3/external/edward.py]: new-line

**Pull request merged**